dbms/src/Flash/Pipeline/Exec/PipelineExec.cpp

```diff
@@ -46,6 +46,41 @@ namespace DB
         return (op_status);                                                                       \
     }
 
+#define HANDLE_AWAITABLE_OP_STATUS(op, op_status)                                                 \
+    switch (op_status)                                                                            \
+    {                                                                                             \
+    case OperatorStatus::NEED_INPUT:                                                              \
+        break;                                                                                    \
+    /* For the io status, the operator needs to be filled in io_op for later use in executeIO. */ \
+    case OperatorStatus::IO:                                                                      \
+        assert(!io_op);                                                                           \
+        assert(op);                                                                               \
+        io_op.emplace((op)->getOp());                                                             \
+    /* For unexpected status, an immediate return is required. */                                 \
+    default:                                                                                      \
+        return (op_status);                                                                       \
+    }
+
+PipelineExec::PipelineExec(
+    SourceOpPtr && source_op_,
+    TransformOps && transform_ops_,
+    SinkOpPtr && sink_op_)
+    : source_op(std::move(source_op_))
+    , transform_ops(std::move(transform_ops_))
+    , sink_op(std::move(sink_op_))
+{
+#define FILL_AWAITABLE(op)                                                   \
```

Comment:
use inline method to replace macro? And can add comments to limit its usage.

Reply:
ok, updated.